redbot/cogs/streams/streams.py

```diff
@@ -714,14 +714,23 @@ async def _stream_alerts(self):
             await asyncio.sleep(await self.config.refresh_timer())
 
     async def _send_stream_alert(
-        self, stream, channel: discord.TextChannel, embed: discord.Embed, content: str = None
+        self,
+        stream,
+        channel: discord.TextChannel,
+        embed: discord.Embed,
+        content: str = None,
+        *,
+        is_schedule: bool = False,
     ):
         m = await channel.send(
             content,
             embed=embed,
             allowed_mentions=discord.AllowedMentions(roles=True, everyone=True),
         )
-        stream._messages_cache.append(m)
+        message_data = {"channel": m.channel.id, "message": m.id}
```

Comment:
Can we include the guild ID in here so that when we pull the channel again later it doesn't have to iterate through all the bots guilds to find the channel?